src/lib.rs

```diff
@@ -19,33 +19,25 @@ fn parse_quoted(input: &str) -> IResult<&str, String> {
     Ok(res)
 }
 
-fn and_parser(i: &str) -> IResult<&str, &str> {
-    tag("&&")(i)
-}
-fn or_parser(i: &str) -> IResult<&str, &str> {
-    tag("||")(i)
-}
-fn and_or_choice(i: &str) -> IResult<&str, &str> {
-    alt((and_parser, or_parser))(i)
-}
 
-fn base_parser(i: &str) -> IResult<&str, BaseOutput, Error<&str>> {
-    let f =  tag("foo");
-    let eq_op  =  tag("==");
+fn base_parser<'a>() -> impl FnMut(&'a str) -> IResult<&'a str, BaseOutput, Error<&'a str>> {
+    let f = tag("foo");
+    let eq_op = tag("==");
```

Comment:
do you work with a linter?? I use webstorm and I'll be happy to configure one

Reply:
yes, there is one linter out there: [clippy](https://github.com/rust-lang/rust-clippy) (this extends cargo check).
You can use [rust plugin](https://plugins.jetbrains.com/plugin/8182-rust), using Clion will allow you to debug and such but you don't have to use it.